lib/improver/blending/weighted_blend.py

```diff
@@ -536,6 +535,11 @@ def process(self, cube, weights=None):
                 if perc_coord and self.mode == "weighted_mean":
                     percentiles = np.array(perc_coord.points, dtype=float)
                     perc_dim, = cube_thres.coord_dims(perc_coord.name())
+                    # The Iris aggregate method moves the collapse coordinate
+                    # to index=-1, so we must adjust perc_dim if moving the
```

Comment:
I was a bit confused by this description - can we be more precise?

"The iris.analysis.Aggregator class moves the collapsed coordinate to index=-1 in initialisation, before the aggregation method is called, which shifts back all later coordinates by 1.  If the collapsed coordinate occurs before the percentile coordinate in the input cube, we need to adjust for this before calling the Aggregator."

If this is too clunky for inline, perhaps expand the "process" docstring.